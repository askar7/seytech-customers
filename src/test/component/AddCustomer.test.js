import React from 'react';
import AddCustomer from '../../components/AddCustomer';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer'; // snapshot testing

describe('AddCustomer tests', () => {
  // snapshot
  test('renders snapshot', () => {
    const app = renderer.create(<AddCustomer />);
    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('simulate a click', () => {
    let app = mount(<AddCustomer />);
    app.find('button').at(0).simulate('click');
    const modalElement = app.find('.modal-content');
    // console.log(modalElement.debug());
    expect(modalElement).toBeTruthy();
  });
});
