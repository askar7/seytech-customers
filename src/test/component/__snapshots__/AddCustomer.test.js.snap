// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCustomer tests renders snapshot 1`] = `
<div>
  <button
    aria-label={null}
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Add new customer
  </button>
</div>
`;
